ArduCopter/mode_guided.cpp

```diff
@@ -493,15 +547,34 @@ void ModeGuided::vel_control_run()
     // set velocity to zero and stop rotating if no updates received for 3 seconds
     uint32_t tnow = millis();
     if (tnow - vel_update_time_ms > GUIDED_POSVEL_TIMEOUT_MS) {
-        if (!pos_control->get_vel_desired_cms().is_zero()) {
-            set_desired_velocity_with_accel_and_fence_limits(Vector3f(0.0f, 0.0f, 0.0f));
-        }
+        guided_vel_target_cms.zero();
+        guided_accel_target_cmss.zero();
         if (auto_yaw.mode() == AUTO_YAW_RATE) {
             auto_yaw.set_rate(0.0f);
         }
+    }
+
+#if AC_AVOID_ENABLED
+    // limit the velocity for obstacle/fence avoidance
+    copter.avoid.adjust_velocity(guided_vel_target_cms, pos_control->get_pos_xy_p().kP(), pos_control->get_max_accel_xy_cmss(), pos_control->get_pos_z_p().kP(), pos_control->get_max_accel_z_cmss(), G_Dt);
+#endif
+
+    // update position controller with new target
+    if (stop_stabilize_vel_xy()) {
+        // set the current commanded xy vel to the desired vel
+        guided_vel_target_cms.x = pos_control->get_vel_desired_cms().x;
```

Comment:
I guess this means if we're not stabilizing_vel that object avoidance doesn't work because it's overwriting the guided_vel_target_cms.x/y values that were just updated by avoid.